cmd/main.go

```diff
@@ -199,10 +215,30 @@ func main() {
 		Cache: cache.Options{
 			SyncPeriod: &syncPeriod,
 		},
-	})
-	if err != nil {
-		setupLog.Error(err, "unable to start manager")
+	}
+	restConfig := ctrl.GetConfigOrDie()
+
+	if isKcp, err = kcpAPIsGroupPresent(restConfig); err != nil {
+		setupLog.Error(err, "to determine if kcp API Group is present")
 		os.Exit(1)
+	} else if isKcp {
+		setupLog.Info("Found KCP APIs, looking up virtual workspace URL")
+		exportConfig, err := restConfigForLogicalClusterHostingAPIExport(context.Background(), restConfig, apiExportName)
+		if err != nil {
+			setupLog.Error(err, "looking up virtual workspace URL")
+			os.Exit(1)
+		}
+		mgr, err = kcp.NewClusterAwareManager(exportConfig, ctrlOptions)
+		if err != nil {
+			setupLog.Error(err, "unable to create kcp aware manager")
+		}
+	} else {
+		setupLog.Info("Did not find KCP APIs, starting as usual")
```

Comment:
maybe the `starting as usual` refers to the history of implementation too much. I would suggest to rephrase it more generically, e.g.: `KCP APIs have not been found. Assuming default k8s cluster`